spec/node/asciidoctor.spec.js

```diff
@@ -48,6 +48,34 @@ describe('Node.js', () => {
     });
   });
 
+  if (asciidoctor.LoggerManager) {
+    describe('Logger', () => {
+
+      it('should warn if part has no sections', () => {
+        const input = `= Book
+:doctype: book
+
+= Part 1
+
+[partintro]
+intro
+`;
+        const defaultLogger = asciidoctor.LoggerManager.getLogger();
+        const memoryLogger = asciidoctor.MemoryLogger.$new();
+        try {
+          asciidoctor.LoggerManager.setLogger(memoryLogger);
+          asciidoctor.convert(input);
+          const errorMessage = memoryLogger.getMessages()[0];
+          expect(errorMessage.severity.toString()).toBe('ERROR');
+          expect(errorMessage.message['text']).toBe('invalid part, must have at least one section (e.g., chapter, appendix, etc.)');
+          expect(errorMessage.message['source_location'].lineno).toBe(8);
```

Comment:
Since the `Cursor` object will be available in the context, we should probably add a wrapper API around this object:

```js
const cursor = Asciidoctor.Cursor.$new();
cursor.getDirectory();
cursor.getFile();
cursor.getPath();
cursor.getLineNumber();
```